### ISSUE TITLE
[test] Improve xml test schema

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3302]
**Is your feature request related to a problem? Please describe.**
Currently there's a few things that are not so great with the test schema
1. the only way to ignore a test is to set the attribute `regressionTest` to false, which is... weird
1. there's no way to execute a single test ("focus" a test). Some IDEs may provide this facility (eclipse), but some don't (intellij), and having an ide-independent way to do this would be nice
1. the attribute `reinitializeRule` of test-code is useless. I can't see a reason *not* to reinitialize the rule.
1. we could remove the pattern from the `source-type` element, as we do the validation at runtime anyway. This means we wouldn't have to update the schema when we add/remove languages, which makes pmd easier to extend 
1. validation of consistency between expected-problems, expected-messages and expected-linenumbers is buggy. Sometimes it throws a validation error, sometimes it's silent. For instance
   ```xml 
        <expected-problems>0</expected-problems>
        <expected-messages>
            <message>1</message>
        </expected-messages>
   ``` 
  throws a validation error, but the following doesn't
  ```xml
        <expected-problems>1</expected-problems>
        <expected-linenumbers>1,2,3</expected-linenumbers>
        <expected-messages>
            <message>1</message>
        </expected-messages>
```

**Describe the solution you'd like** Update the schema and its parser to support `focus` and `disabled` attributes, deprecate useless things

**Describe alternatives you've considered**

**Additional context**
Some other things that we should keep in mind and fix later:
1. the `<rule-property>` element has ad-hoc syntax, instead of just reusing the syntax for `<property>` that's from the ruleset schema. This will be awkward when the XSD for rulesets is updated (#1432)
2. the `useAuxClasspath` attribute doesn't look useful. Related to #2649

**Comment by adangel** [comment]
The following points are addressed by #3976: 1 (regressionTest), 4 (source-type), 5 (validation of expected-problems, expected-linenumbers, expected-messages)

TODO is: 2 (focus), 3 (reinitializeRule)

**Comment by adangel** [comment]
I noticed that point 3 is also done: reinitalizeRule and auxclasspath are deprecated

**Comment by oowekyala** [comment]
@adangel I will follow up with a fix for point 2, and update the release notes with refs to 3 as well.
